Move view note link and minimize the margin

amdomanska opened the pull request:
https://github.com/DOAJ/doajPM/issues/2904#issuecomment-998242478

![image](https://user-images.githubusercontent.com/8298769/147746061-4c0b30ea-406d-4f33-b560-444a9ffe5e67.png)

@richard-jones simple js change.

@sssoz not sure whether that's what was meant by "close up the space", let me know if not. Also - I couldn't find a good place to add the css (text area has in general margin-bottom: 0.75 so to close up the space I just added -0.75 to the "view note" link). Could you move the inline style to some better place? Or maybe it can stay like this?

richard-jones commented on portality/static/js/formulaic.js:
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
             this.init = function() {
                 var viewClass = edges.css_classes(this.ns, "view");
                 var closeClass = edges.css_classes(this.ns, "close");
+                var viewDivClass = edges.css_classes(this.ns, "viewDiv")
```
This class doesn't seem to get used

richard-jones commented on portality/static/js/formulaic.js:
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
                 var viewClass = edges.css_classes(this.ns, "view");
                 var closeClass = edges.css_classes(this.ns, "close");
+                var viewDivClass = edges.css_classes(this.ns, "viewDiv")
+                this.divs = $("div[name='" + this.fieldDef["name"] + "__group']").find("textarea");
```
Are we sure that replacing the `this.divs` member variable with a set of `textarea`s doesn't have any side effects in other methods that use the divs?

amdomanska replied:
You're right, it shouldn't be done this way. I will simply use a new local var.

sssoz commented on portality/static/js/formulaic.js:
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
                     var note = $("#" + this.fieldDef["name"] + "-" + i + "-note");
 
-                    container.append(`<div><a href="#" class="` + viewClass + `">view note</a>
+                    $(`<button class="button button--cta ` + viewClass + `" style="margin: 0 1rem 1rem 0;">View note</button>
```
@amdomanska I’ve made some slight modifications to the styling here (explained in here: https://github.com/DOAJ/doajPM/issues/2904#issuecomment-1008761062)

sssoz commented:
@amdomanska I’ve merged this branch to mine, `feature/2904-notes-fix` (in https://github.com/DOAJ/doaj/pull/2017)

amdomanska commented:
@richard-jones I have addressed your comments.
@sssoz After Richard will approve my latest changes feel free to merge my branch to yours again to have the most up to date version. Then we can close that pull request and leave just yours.

sssoz commented:
Thanks @amdomanska ! 🤦‍♀️ I lost track of my own PRs and branches and didn’t realise I didn’t have a PR for this — it was just a blank, commit-less branch! Sorry for the confusion. We can simply keep things here in this PR
